client/client_test.go

```diff
@@ -1632,6 +1632,38 @@ func TestPublishTargetsDelgationSuccessLocallyHasRoles(t *testing.T) {
 		[]string{"targets/a/b"})
 }
 
+// If a changelist specifies a particular role to push targets to, and the role
+// is present, publish will write to that role only.  The targets keys are not
+// needed.
+func TestPublishTargetsDelgationNoTargetsKeyNeeded(t *testing.T) {
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	assert.NoError(t, err, "failed to create a temporary directory: %s", err)
+	defer os.RemoveAll(tempBaseDir)
+
+	gun := "docker.com/notary"
+	ts := fullTestServer(t)
+	defer ts.Close()
+
+	repo, _ := initializeRepo(t, data.ECDSAKey, tempBaseDir, gun, ts.URL, false)
+	delgKey, err := repo.CryptoService.Create("targets/a", data.ECDSAKey)
+	assert.NoError(t, err, "error creating delegation key")
+
+	for _, delgName := range []string{"targets/a", "targets/a/b"} {
+		assert.NoError(t,
+			repo.AddDelegation(delgName, 1, []data.PublicKey{delgKey}, []string{""}),
+			"error creating delegation")
+	}
+	assert.NoError(t, repo.Publish())
+
+	// remove targets key - it is not even needed
+	targetsKeys := repo.CryptoService.ListKeys(data.CanonicalTargetsRole)
+	assert.Len(t, targetsKeys, 1)
+	assert.NoError(t, repo.CryptoService.RemoveKey(targetKeys[0]))
```

Comment:
Yep, thanks for the catch!  :)  that was the error - I think I deleted it while switching windows :(